overlord/devicestate/devicestate.go

```diff
@@ -1239,6 +1239,12 @@ func Remodel(st *state.State, new *asserts.Model, localSnaps []*snap.SideInfo, p
 		return nil, fmt.Errorf("cannot remodel from grade %v to grade %v", current.Grade(), new.Grade())
 	}
 
+	if new.Base() == "" || new.Base() == "core" {
```

Comment:
strictly speaking Base == "core" is not supported at all because Base != "" means UC18 using snapd with a boot base but core itself does not work in that setting, it doesn't contain the right bits to be used as a boot base the UC18 way

Reply:
So I can remove the checks for `new.Base() == "core"` and `current.Base() == "core"`?

Reply:
yes think so